src/main/java/com/pinterest/secor/uploader/HadoopS3UploadManager.java

```diff
@@ -63,7 +82,14 @@ public HadoopS3UploadManager(SecorConfig config) {
             @Override
             public void run() {
                 try {
-                    FileUtil.moveToCloud(localLogFilename, logFileName);
+                    if (customTopicsNamesMap != null) {
+                    	String logFileNameCustomTopicName = logFileName;
+                    	logFileNameCustomTopicName = logFileNameCustomTopicName.replace(topicName,
```

Comment:
This is error prone, you might replace more than the topic name (if the topicName is also present in other parts of your path).

I think the change needs to be in localPath.getLogFilePath where you can precisely replacing just the topicName in the path.

Reply:
Thank you @HenryCaiHaiying for advice. I agree with you. I will precisely build the custom path in getLogFilePath method by using SecorConfig mConfig to obtain map parameters of custom names.